src/Kiota.Builder/CodeDOM/CodeFunction.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Linq;
+
+namespace Kiota.Builder;
+
+/// <summary>
+/// Represents global functions for languages that support both local functions to classes and global functions like TypeScript for instance.
+/// </summary>
+public class CodeFunction : CodeBlock<BlockDeclaration, BlockEnd>
+{
+    public override string Name
+    {
+        get
+        {
+            return OriginalLocalMethod.Name;
```

Comment:
Defensive programming as `OriginalLocalMethod` might return null as it calls `FirstOrDefault`

Reply:
thanks for pointing this out. Storing the original method outside of the child items (didn't make sense to put it there because we never want to represent both), and the defensive programing on the constructor + the private access to the setter now prevent it from ever being nullable